clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3025,6 +3048,11 @@ private int processKafkaDataMessage(
         }
         keyLen = keyBytes.length;
 
+        if (recordTransformer != null) {
+          Lazy<byte[]> lazyKey = Lazy.of(() -> keyBytes);
+          GenericRecord keptRecord = recordTransformer.delete(lazyKey, producedPartition);
```

Comment:
It seems keptRecord isn't used?  We'll need to carry the output of the key/partition over to removeFromStorageEngine in order for it to properly get cleaned up.

Reply:
I think I may be misunderstanding what the delete method in the recordTransformer should do. My thought process was that the delete method would be called in the recordTransformer which handles the deletion of the key that is seperate (maybe internally or in the outPutCollector we discussed?) from the removeFromStorageEngine function, and is defined to maybe return something if the user wants to hold onto to the record for some reason. So right now I wasn't sure what to do with keptRecord. But now I am thinking that the key/partition needs to be transformed in order to delete it from the right place in the storage engine, and one can choose to not remove it from the storage engine by letting it return the null default value.